### PR TITLE
MAINT: remove u-prefix for former Unicode strings

### PR DESCRIPTION
**Pull request by mwtoews**
This PR removes the "u" prefix that is a relic of Python 2's [Unicode Strings](https://docs.python.org/2/tutorial/introduction.html#unicode-strings), which is redundant since Python 3 (and was briefly removed until Python 3.3 via [PEP 414](https://peps.python.org/pep-0414/)).

Xref #8832

### CODE REVIEW

File: numpy/core/tests/test_regression.py [hunk 1]
```diff
@@ -1396,28 +1396,28 @@ def test_object_array_to_fixed_string(self):
...
         # gh-15790
-        a = np.array([u'\x80'] * 129, dtype='U3')
+        a = np.array(['\x80'] * 129, dtype='U3')
         assert_raises(UnicodeEncodeError, np.array, a, 'S')
         b = a.reshape(3, 43)[:-1, :-1]
         assert_raises(UnicodeEncodeError, np.array, b, 'S')
 
     def test_mixed_string_unicode_array_creation(self):
-        a = np.array(['1234', u'123'])
+        a = np.array(['1234', '123'])
```
Comment (rossbar) [inline numpy/core/tests/test_regression.py:1412]: From the test name, this one actually looks like the first instance in the array was actually *intended* to be a Python2-style byte string. IIUC I think the test should be updated to include the `b` prefix for the array elements that are not marked with `u`.

**Review by rossbar — COMMENTED** [review]
Generally LGTM - a nice straightforward cleanup, thanks @mwtoews . Just one test that looks like it might need updating in the other direction (i.e. making some instances `bytes` explicitly).

File: numpy/core/tests/test_regression.py (hunk 1 above)
Comment (mwtoews) [reply numpy/core/tests/test_regression.py]: good catch, I've update this test to interleave string and byte objects

**Comment by rossbar** [comment]
Thanks @mwtoews !

**Pull request merged**